src/stories/TextInput.stories.tsx

```diff
@@ -162,66 +162,111 @@ export const WithTrailingAction = (args: FormControlArgs<TextInputProps>) => {
   )
 }
 
-export const WithLoadingIndicator = (args: FormControlArgs<TextInputProps>) => (
-  <Box as="form" sx={{p: 3}}>
-    <h3>No visual</h3>
-    <Box mb={2}>
-      <TextInput value="auto" {...args} />
-    </Box>
-    <Box mb={2}>
-      <TextInput value="leading" {...args} loaderPosition="leading" />
-    </Box>
-    <Box mb={5}>
-      <TextInput value="trailing" {...args} loaderPosition="trailing" />
-    </Box>
+export const WithLoadingIndicator = (args: FormControlArgs<TextInputProps>) => {
+  const {parentArgs, labelArgs} = getFormControlArgsByChildComponent(args)
+  return (
+    <Box as="form" sx={{padding: 3}}>
+      <h3>No visual</h3>
 
-    <h3>Leading visual</h3>
-    <Box mb={2}>
-      <TextInput leadingVisual={CalendarIcon} {...args} value="auto" />
-    </Box>
-    <Box mb={2}>
-      <TextInput leadingVisual={CalendarIcon} {...args} loaderPosition="leading" value="leading" />
-    </Box>
-    <Box mb={5}>
-      <TextInput leadingVisual={CalendarIcon} {...args} loaderPosition="trailing" value="trailing" />
-    </Box>
+      <Box mb={2}>
+        <FormControl {...parentArgs}>
```

Comment:
@joshblack I literally copy-pasted the default story struvture. I didn't do any optimisation of anything mainly because we have [a renaming stories task](https://github.com/github/primer/issues/1488) and I thought we could handle some refactor there if we needed to. Let me know what you think!